src/main/mediaPermissions.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * Vesktop, a desktop app aiming to give you a snappier Discord Experience
+ * Copyright (c) 2023 Vendicated and Vencord contributors
+ */
+
+import { session, systemPreferences } from "electron";
+
+export function registerMediaPermissionsHandler() {
+    if (process.platform !== "darwin") return;
+
+    session.defaultSession.setPermissionRequestHandler((_webContents, permission, callback, details) => {
+        if (permission === "media") {
+            if (details.mediaTypes?.includes("audio")) {
+                systemPreferences.askForMediaAccess("microphone").then(callback);
+            }
+            if (details.mediaTypes?.includes("video")) {
```

Comment:
```suggestion
            else if (details.mediaTypes?.includes("video")) {
```

Reply:
actually, as pointed out by lumap, it might be both at once. in any case, the default case should call callback(true)

Reply:
Uh, my current implementation supports both at once. It's not if-else, it's if and if. Default case being callback(true) makes no sense to me, could you elaborate?

Reply:
Hm, I guess that makes sense.

Reply:
As for the if blocks, your suggestion of using `else if` instead of `if` makes the permission requests mutually exclusive. My initial implementation makes sure that both can be requested at the same time.